kimchi/src/proof.rs

```diff
@@ -735,85 +742,83 @@ pub mod caml {
+                    .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
+                lookup_table: pe
+                    .lookup_table
+                    .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
+                lookup_sorted: pe
+                    .lookup_sorted
+                    .iter()
+                    .map(|x| {
+                        x.as_ref().map(|x| {
+                            x.map_ref(&|x| x.clone().into_iter().map(Into::into).collect())
                         })
-                        .collect::<Vec<_>>(),
-                    runtime_lookup_table: pe
-                        .runtime_lookup_table
-                        .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
-                    runtime_lookup_table_selector: pe
-                        .runtime_lookup_table_selector
-                        .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
-                    xor_lookup_selector: pe
-                        .xor_lookup_selector
-                        .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
-                    lookup_gate_lookup_selector: pe
-                        .lookup_gate_lookup_selector
-                        .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
-                    range_check_lookup_selector: pe
-                        .range_check_lookup_selector
-                        .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
-                    foreign_field_mul_lookup_selector: pe
-                        .foreign_field_mul_lookup_selector
-                        .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
-                },
-            )
+                    })
+                    .collect::<Vec<_>>(),
+                runtime_lookup_table: pe
+                    .runtime_lookup_table
+                    .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
+                runtime_lookup_table_selector: pe
+                    .runtime_lookup_table_selector
+                    .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
+                xor_lookup_selector: pe
+                    .xor_lookup_selector
+                    .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
+                lookup_gate_lookup_selector: pe
+                    .lookup_gate_lookup_selector
+                    .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
+                range_check_lookup_selector: pe
+                    .range_check_lookup_selector
+                    .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
+                foreign_field_mul_lookup_selector: pe
+                    .foreign_field_mul_lookup_selector
+                    .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
+            };
+
+            (first, second)
```

Comment:
Not sure I like this distinction

Reply:
Let me explain: 
1. The original function's `return` statement was `80` lines, including nested expressions and a lot of commas, so it was very confusing as to what is the return type at all. I like it when you can fast-forward to the end of the function and see what it does.
2. The `From::` conversions in this whole function are, arguably, quite non-intuitive -- it took me quite some time to understand how `map` (being a custom function! and not a generic iterator) is implemented. Figuring this out is further complicated by `From::` instances being picked implicitly, so it takes time to find them. This is why I explicitly typed `first` so that one can see by example what's happening here.

TLDR I just spent 15 minutes trying to figure out what's going on here and thought this might help...